components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -710,14 +716,14 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 								}
 							}
 						}
-						if !imagestreamFound {
-							log.Error(nil, "Imagestream not found in main controller namespace", "imageSelected", imageSelected[0], "tag", imageSelected[1], "namespace", namespace)
-						}
+					}
+					if !imagestreamFound {
+						log.Error(nil, "Imagestream not found in main controller namespace", "imageSelected", imageSelected[0], "tag", imageSelected[1], "namespace", namespace)
 					}
 				}
 			}
 			if !containerFound {
-				log.Error(nil, "No container found matching the notebook name", "notebookName", notebook.Name)
+				return fmt.Errorf("no container found matching the notebook name %s", notebook.Name)
```

Comment:
@jstourac this causes the webhook to refuse to admit the notebook CR, which is imo the correct behavior, and it is not completely different from what this was doing before (when webhook panicked on the nil value, it would also refuse to admit)